pandas/tests/io/test_excel.py

```diff
@@ -58,11 +51,16 @@ def ignore_xlrd_time_clock_warning():
         yield
 
 
-@td.skip_if_no('xlrd', '1.0.0')
-class ReadingTestsBase:
-    # This is based on ExcelWriterBase
-
-    @pytest.fixture(autouse=True, params=['xlrd', None])
+@pytest.mark.parametrize("ext", ['.xls', '.xlsx', '.xlsm'])
+class TestReaders:
+
+    @pytest.fixture(autouse=True, params=[
+        # Add any engines to test here
+        pytest.param('xlrd', marks=pytest.mark.skipif(
```

Comment:
Might be a cleaner way to mark these but in any case this is more or less what I see being easiest long term as we look at add openpyxl into the reading mix. Any reader can be parametrized here with this fixture and automatically cover all of the reading tests.

I'll have to move or split some tests out of the base class here that truly do test xldd-specific behavior but leaving that for another follow up

Reply:
That's an option but once we start adding more engines with different file extensions (ex: openpyxl won't handle xlrd, odfpy won't be excel files) that would generate a ton of skips. One option to reduce that would be to change the scope here to a higher level, but I think it makes sense to defer that or whatever the ultimate change is for now

Reply:
ok, for now then as will also depend on how engine=None will be tested if that's changed.